fields.go

```diff
@@ -28,46 +30,42 @@ func appendFields(dst []byte, fields interface{}, stack bool) []byte {
 		kv := make([]interface{}, 2)
 		for _, key := range keys {
 			kv[0], kv[1] = key, fields[key]
-			dst = appendFieldList(dst, kv, stack)
+			dst = appendFieldList(dst, kv, stack, ctx, hooks)
 		}
 	}
 	return dst
 }
 
-func appendObject(dst []byte, obj LogObjectMarshaler) []byte {
-	e := newEvent(nil, 0)
+func appendObject(dst []byte, obj LogObjectMarshaler, stack bool, ctx context.Context, hooks []Hook) []byte {
+	e := newEvent(LevelWriterAdapter{io.Discard}, DebugLevel, stack, ctx, hooks)
```

Comment:
Can't use `Logger`'s `scratchEvent` here because this is not bound to an `*Event`.